### PR TITLE
fix: use max spendable amount to sweep balance

### PR DESCRIPTION
**Pull request by hrishikeshjain**
TICKET: COIN-2432

Currently `/sweep` api endpoint throws insufficent funds error because while building transaction we check for amount + fee and amount that we were sending was spendable balanceso it will always fail . So updated it to maxSpendable balance which returns amount subtrcating fee.

### CODE REVIEW

File: modules/sdk-core/src/bitgo/wallet/wallet.ts [hunk 1]
```diff
@@ -938,14 +938,15 @@ export class Wallet implements IWallet {
         );
       }
 
-      const value = this.spendableBalanceString();
-      if (_.isUndefined(value) || value === '0') {
+      const value = await this.maximumSpendable();
```
Comment (gianchandania) [inline modules/sdk-core/src/bitgo/wallet/wallet.ts:941]: Add tests for the change

Comment (hrishikeshjain) [reply modules/sdk-core/src/bitgo/wallet/wallet.ts:941]: It looks like we cant write test case for sweep functionality, @gianchandania any idea if theres a way?

File: modules/sdk-core/src/bitgo/wallet/wallet.ts [hunk 2]
```diff
@@ -938,14 +938,15 @@ export class Wallet implements IWallet {
...
+      const maximumSpendable = new BigNumber(value.maximumSpendable).toString();
+      if (_.isUndefined(value) || maximumSpendable === '0') {
         throw new Error('no funds to sweep');
       }
       (params as any).recipients = [
         {
           address: params.address,
-          amount: value,
+          amount: maximumSpendable,
```
Comment (mohammadalfaiyazbitgo) [inline modules/sdk-core/src/bitgo/wallet/wallet.ts:949]: you don't need `lodash` to do a simple `!== undefined` check. Also why convert `maximumSpendable` to a string to perform a numerical check? Why not https://mikemcl.github.io/bignumber.js/#isPos ?

File: modules/sdk-core/src/bitgo/wallet/wallet.ts [hunk 3]
```diff
@@ -938,14 +938,15 @@ export class Wallet implements IWallet {
...
       }
 
-      const value = this.spendableBalanceString();
-      if (_.isUndefined(value) || value === '0') {
+      const value = await this.maximumSpendable();
+      const maximumSpendable = new BigNumber(value.maximumSpendable).toString();
+      if (_.isUndefined(value) || maximumSpendable === '0') {
         throw new Error('no funds to sweep');
       }
       (params as any).recipients = [
```
Comment (mohammadalfaiyazbitgo) [inline modules/sdk-core/src/bitgo/wallet/wallet.ts:946]: nit: if you can, could you please refactor this to avoid `any` ?

**Review by mohammadalfaiyazbitgo — CHANGES_REQUESTED** [review]

File: modules/sdk-core/src/bitgo/wallet/wallet.ts (hunk 1 above)
Comment (mohammadalfaiyazbitgo) [reply modules/sdk-core/src/bitgo/wallet/wallet.ts:941]: > It looks like we cant write test case for sweep functionality

Can you elaborate on why?

File: modules/sdk-core/src/bitgo/wallet/wallet.ts (hunk 2 above)
Comment (hrishikeshjain) [reply modules/sdk-core/src/bitgo/wallet/wallet.ts]: updated 👍

File: modules/sdk-core/src/bitgo/wallet/wallet.ts (hunk 1 above)
Comment (hrishikeshjain) [reply modules/sdk-core/src/bitgo/wallet/wallet.ts]: Added tests 😄

**Comment by hrishikeshjain** [comment]
@mohammadalfaiyazbitgo can I get a re-review please

**Review by mohammadalfaiyazbitgo — APPROVED** [review]

**Pull request merged**